master/internal/db/postgres_proto.go

```diff
@@ -38,18 +39,32 @@ func protoParser(rows *sqlx.Rows, val interface{}) error {
 	if !ok {
 		return errors.Errorf("invalid type conversion: %T is not a Protobuf message", val)
 	}
+
+	cols, err := rows.ColumnTypes()
+	if err != nil {
+		return err
+	}
+	colTypes := make(map[string]string)
+	for _, col := range cols {
+		colTypes[col.Name()] = col.DatabaseTypeName()
+	}
+
 	dest := make(map[string]interface{})
-	if err := rows.MapScan(dest); err != nil {
+	if err = rows.MapScan(dest); err != nil {
 		return errors.Wrap(err, "error reading row from database")
 	}
 	for key, value := range dest {
 		switch parsed := value.(type) {
 		case []byte:
 			var marshaled interface{}
-			if err := json.Unmarshal(parsed, &marshaled); err != nil {
+			if err = json.Unmarshal(parsed, &marshaled); err != nil {
 				return errors.Wrapf(err, "error parsing field: %s", key)
 			}
 			dest[key] = marshaled
+		case time.Time:
+			if colTypes[key] == "DATE" {
```

Comment:
if the column type for this key isn't `DATE`, it seems like we should error or something, not just continue without scanning a value.